providers/infrastructure-vsphere/v1.5.1/cconly/base.yaml

```diff
@@ -1038,7 +1038,7 @@ spec:
                     value: "20"
                   - name: vip_retryperiod
                     value: "4"
-                  image: {{(index (index .TKR_DATA .builtin.controlPlane.version).kubernetesSpec "kube-vip").imageRepository}}/kube-vip:{{(index (index .TKR_DATA .builtin.controlPlane.version).kubernetesSpec "kube-vip").imageTag}}
+                  image: {{(index .TKR_DATA .builtin.controlPlane.version).kubernetesSpec.imageRepository}}/kube-vip:{{(index (index .TKR_DATA .builtin.controlPlane.version).kubernetesSpec "kube-vip").imageTag}}
```

Comment:
I think the ideal behavior would be:
1. Check for `TKR_DATA.value.(k8s-version).kubernetesSpec.kube-vip.imageRepository`. If set, use that.
2. Otherwise, use `TKR_DATA.value.(k8s-version).kubernetesSpec.imageRepository`.

We missed a scenario by always doing (1). I worry we may miss some other scenario by always doing (2).

(I don't _know of_ any such scenario though; I just worry.)

Is the extra complexity of the conditional logic worth considering?

Reply:
I will try to look at the behavior of the other images and see how they handle it, but totally 💯 to checking if `kube-vip` has that value set, then use that otherwise use the top level image repository info.

Reply:
This would be the new sentence
```
image: {{ if (index (index .TKR_DATA .builtin.controlPlane.version).kubernetesSpec "kube-vip").imageRepository }}{{(index (index .TKR_DATA .builtin.controlPlane.version).kubernetesSpec "kube-vip").imageRepository}}{{ else }}{{(index .TKR_DATA .builtin.controlPlane.version).kubernetesSpec.imageRepository}}{{ end }}/kube-vip:{{(index (index .TKR_DATA .builtin.controlPlane.version).kubernetesSpec "kube-vip").imageTag}}
```
Readability is bad. I think we could get confirmation from @ivan whether these <component>.imageRepository are really needed

Reply:
https://github.com/vmware-tanzu/tanzu-framework/blob/949c6d04ba7496537b4192fc129f65fe0b1b2a3d/providers/infrastructure-vsphere/v1.5.1/cconly/base.yaml#L951-L970
None of the above do the conditional check

Reply:
Good point, @srm09. If we're not doing the conditional check in other places, perhaps for now we should consistently not do the conditional check.

If there's potentially a use case for conditional logic we could explore that separately/holistically.

Reply:
Idea from @imikushin: Instead of putting the conditional logic here on the "consumer" side, teach the the tkr webhook to cascade the top-level {{imageRepository}} value down to each component when its unset at the component level.

That seems like a good, consistent medium-term fix.

Reply:
See #4298 for the above.